lib/Runtime/Library/WasmLibrary.cpp

```diff
@@ -307,6 +307,42 @@ namespace Js
 
     void WasmLibrary::WasmLoadIndirectFunctionTables(Wasm::WasmModule * wasmModule, ScriptContext* ctx, Var** indirectFunctionTables, Var* localModuleFunctions, Var* importFunctions)
     {
+        ArenaAllocator alloc(_u("WasmTempIndirectTable"), ctx->GetThreadContext()->GetPageAllocator(), Js::Throw::OutOfMemory);
```

Comment:
Yes, I did think about something like that too. I am wondering if it would be better/cleaner.
I can give it a try